Project feature

echennau opened the pull request:
Added prop-types package.
Missing images for View Demo and other items on the ProjectDetails components.

![image](https://user-images.githubusercontent.com/37466516/95781611-0d667680-0c83-11eb-9406-c7a19f0c60c7.png)
![image](https://user-images.githubusercontent.com/37466516/95781646-22430a00-0c83-11eb-9f04-1deeee1be8c4.png)

echennau committed:
Created project component

echennau committed:
fixed eslint

echennau committed:
Merge branch 'master' into project

echennau committed:
split main project component into seperate component files, added comment section component

RonaldRusso94 committed:
Update tailwind.config.js

echennau committed:
added prop-types, improved styling

Added prop-types module to follow eslint rules, created CommentCard component, created project page, and improved resize styling.

echennau committed:
Merge branch 'project' of https://github.com/RonaldRusso94/rummmble-frontend into project

RonaldRusso94 reviewed (commented):
- Lots of unnecessary css classes. 
- Width percentages unorganized
- Overall it matches the design very well.
- CommentForm Needs work

I have cleaned up and added small fixes to some components. Please review my commit
Good work!

RonaldRusso94 committed:
Css fix & cleanup

echennau committed:
Merge branch 'master' into project

3than0ls commented:
Thanks for your feedback! I have a few questions. Vercel is telling me Deployment has failed, and I'm not really sure how to solve that (every link I click takes me to a 404 page), so how I'm not sure how to fix that. Next is I'm not too sure how I'm supposed to style the CommentForm, and if you have any suggestions on how you want it to look I can change it. Finally, if there's anything I'm (likely) doing wrong, please tell me (especially if it relates to GitHub, because I'm still not super familiar with how this works).

echennau committed:
updated icons

echennau committed:
created Icon component

icon SVG paths and names are from zondicons

RonaldRusso94 committed:
Merge branch 'master' into project

RonaldRusso94 commented:
Last thing is it needs a mobile view. On mobile have it display the project, project details, comment form, comments

echennau committed:
updated mobile css

added a lot of lines of css because of the way tailwind handles mobile resizing, go through it if you want, there may be some unnecesary classes.
A possible issue is description card text  overflow that I talk about in the ProjectDisplay file
added a feature that allows you to minimize comments

echennau committed:
updated props and proptypes

echennau committed:
fixed errors missing keys and hydration warning
